pkg/machine/e2e/machine_test.go

```diff
@@ -169,3 +183,34 @@ func copySparse(dst io.WriteSeeker, src io.Reader) (retErr error) {
 	_, err := io.Copy(spWriter, src)
 	return err
 }
+
+func setTmpDir(value string) (string, error) {
+	switch {
+	case runtime.GOOS != "darwin":
+		tmpDir = value
+	case len(value) >= 22:
+		return "", errors.New("path length should be less than 22 characters")
+	case value == "":
+		return "", errors.New("TMPDIR cannot be empty. Set to directory mounted on podman machine (e.g. /private/tmp)")
```

Comment:
I'm okay with setting TMPDIR automatically to `/private/tmp` in this instance

Reply:
This seems to be another Issue.
```
% export TMPDIR=""
% podman machine init --now
Looking up Podman Machine image at quay.io/podman/machine-os:5.1 to create VM
Getting image source signatures
Copying blob 5ab0b6fcee8a done   | 
Copying config 44136fa355 done   | 
Writing manifest to image destination
5ab0b6fcee8aba3f303ad47032c77ffa6894a25f8bc3645b3567860cd7dddf39
Extracting compressed file: podman-machine-default-arm64.raw: done  
Machine init complete
Starting machine "podman-machine-default"
Error: unable to connect to "gvproxy" socket at "podman/podman-machine-default-gvproxy.sock"
```

The following code will set an empty path if `TMPDIR=“”`. So the gvproxy path will be inappropriate.
If `unset TMPDIR` is executed, /tmp is set and it works.

https://github.com/containers/podman/blob/c9644ebccf14309a77769cba00833cd139509e4a/pkg/machine/env/dir_darwin.go#L5-L11